server/gitrest/Dockerfile

```diff
@@ -2,37 +2,83 @@
 # Licensed under the MIT License.
 # DisableDockerDetector "No feasible secure solution for OSS repos yet"
 
-FROM node:18.17.1-buster-slim@sha256:cb2a746612c2564d3bd0f871174618337af9d0f4d895df6a623fd7a69ca6e5bd AS base
+FROM node:18.17.1-buster-slim@sha256:cb2a746612c2564d3bd0f871174618337af9d0f4d895df6a623fd7a69ca6e5bd AS runnerbase
+
+ARG SETVERSION_VERSION=dev
+ENV SETVERSION_VERSION=$SETVERSION_VERSION
+
+ARG SETVERSION_CODEVERSION=dev
+ENV SETVERSION_CODEVERSION=$SETVERSION_CODEVERSION
+
+ARG INTERDEPENDENCY_RANGE=^
+ENV INTERDEPENDENCY_RANGE=$INTERDEPENDENCY_RANGE
+
+ARG RELEASE_GROUP=gitrest
+ENV RELEASE_GROUP=$RELEASE_GROUP
+
+ARG BUILD_ROOT=/usr/FluidFramework
+ARG GITREST_ROOT=$BUILD_ROOT/server/gitrest
 
 # Add Tini
-ENV TINI_VERSION v0.18.0
+ENV TINI_VERSION=v0.18.0
 ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
 RUN chmod +x /tini
 
-# Copy over and build the server
-WORKDIR /home/node/server
+FROM runnerbase AS base
+
+# Copy over and build the server. We use the same directory structure as outside of the docker container to ensure flub works smoothly.
+WORKDIR $GITREST_ROOT
 
 COPY package*.json ./
 COPY pnpm*.yaml ./
+COPY .npmrc ./
+COPY lerna.json ./
 COPY scripts/*.* ./scripts/
 COPY .npmrc ./
 
 COPY packages/gitrest/package*.json packages/gitrest/
 COPY packages/gitrest-base/package*.json packages/gitrest-base/
 
+# Configuring PNPM_HOME allows reusing the host container's pnpm cache by mounting it to
```

Comment:
Should this read "the container's host's pnpm cache"? But even if so, I'm not very clear how we achieve that if that's the case (the docs for `RUN --mount=type-cache` don't suggest to me that's what happens by default with no `from/source` specified). Not sure I get what this accomplishes in general otherwise.

Reply:
Yeah the comment here is bogus. The rationale for including this line is mostly that I cargo-culted several generic improvements suggested by pnpm's [working with docker docs](https://pnpm.io/docker). Configuring `PNPM_HOME` to a well-known location is benign. I believe adding it to the path is only necessary to leverage globally installed tools.

You're right that setting up the cache mount ahead of pnpm install doesn't reuse the container host's cache as is suggested here which is obvious on CI, but it does speed up the incremental docker build for local development.

I can either remove the inclusion of `pnpm_home` in the path or just update this comment to reflect reality--your call :)

Reply:
I like the improvement, updating the comment seems preferable.